bloom-works/coram-uk-child-welfare-playbook
# Test GraphQL access to csv files

## Pull Request Background
The purpose of this PR is to test the feasibility of using `csv` files to update information for this project. I used the AW data that was in the airtable and created a `csv` file for it. I then uploaded this file to the project and tested/was able to access the information using GraphQL.

### What Changed
- Added `gatsby-transformer-csv` to package dependencies and updated gatsby configuration for this new plugin
- Created a data folder to hold `csv` files and added a temporary `AWData.csv` file for testing purposes
- Created a `useAWDataCSV` hook to access the data via GraphQL 
- Added a `console.log` to the Index page(located at `index.tsx`) that prints out the new `csv` data located in the new `src/data` folder.

### How to Test This Work
- Run `npm ci` and then `npm start` to spin up the project.
- Visit the home/index page and open up the console. There you should see the results of the `csv` data being printed successfully. 
- If you would like to do further testing, you can upload a new `csv` file and create a new hook. Alternatively, you can change the data that's currently in the `AWData.csv` file and check that the altered data updates in the console.

### Additional Notes
I am trying to figure out why the `csv` data is nested twice under `awData` (right now the `console.log` requires us to access the data via `awData.awData.allAWDataCsv`) I am not sure where it's getting the extra `awData prop` but trying to figure that out. Otherwise, the transformer works well for csv's.
